terminus/commands.py

```diff
@@ -74,13 +76,21 @@ def on_text_command(self, view, name, args):
             return ("terminus_paste", None)
         elif name == "paste_from_history":
             return ("terminus_paste_from_history", None)
+        elif name == "paste_selection_clipboard":  # Linux middle paste
```

Comment:
why don't you just send the current clipboard to the terminal?

Reply:
Because `g_clipboard_history` is using `sublime.get_clipboard()` and it does not contain the selected text (the clipboard is only populated when triggering an explicit copy (e.g. `ctrl+c`).
Sublime API does not seem to offer a way to recover the selection_clipboard, only `paste_selection_clipboard`.

Reply:
Thanks for the explanation. I think I have misunderstood the functionality of the middle click.

Reply:
For your defense it's a X11 very specific thing :) (https://www.jwz.org/doc/x-cut-and-paste.html) ... works flawlessly in sublime view, but not caught when using Terminus.

Reply:
> works flawlessly in sublime views

Because of [this](https://github.com/randy3k/Terminus/blob/ac6f8e113191348dfbe1afa523b6615d4e48c45d/terminus/commands.py#L53) undoing the paste in Terminus? If this is the case, we should just handle `paste_selection_clipboard` here as well as `insert`.

Reply:
> For your defense it's a X11 very specific thing :) (https://www.jwz.org/doc/x-cut-and-paste.html)

Ya, a longtime Mac user here (and I don't even own a mouse).

Reply:
Mmmh I tried disabling `soft_undo` but doesn't help :(, when I middle paste, the text appears, but isn't taken into account when hitting enter: the command/args are empty in `command, args, _ = view.command_history(0)`.